ConstructionPreferences/Source/ConstructionPreferences/Private/ConstructionPreferencesModule.cpp

```diff
@@ -43,36 +44,24 @@ void FConstructionPreferencesModule::StartupModule() {
 #if !WITH_EDITOR
 	SUBSCRIBE_METHOD(UFGBuildGunStateBuild::SetActiveRecipe, [this](auto& scope, UFGBuildGunStateBuild* self, TSubclassOf< class UFGRecipe > recipe)
 		{
-			if (recipe)
-			{
+			if (!IsValid(recipe)) {
+				return;
+			}
+			if (UCP_GameInstanceModule* CP_Instance = UCP_GameInstanceModule::Get(GWorld)) {
```

Comment:
`GWorld` could be replaced with `self`, I didn't look too closely at the build gun states but they implement `GetWorld()` (they can because they're bound to a build gun actor)